src/time/YamlDateTime.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_TIME_YAMLDATETIME_H
+#define SQUID_SRC_TIME_YAMLDATETIME_H
+
+#include <iosfwd>
+#include <sys/time.h>
+
+namespace Time {
+
+/// Output onto an ostream a yaml-formatted datetime string (UTC)
+///  see https://yaml.org/type/timestamp.html
+class YamlDateTime
+{
+public:
+    YamlDateTime(const struct timeval &tv) : tv_(tv) {};
```

Comment:
Please avoid implicit conversions by default:

```suggestion
    explicit YamlDateTime(const struct timeval &tv): tv_(tv) {};
```